### ISSUE TITLE
`writeToDisk` crashes if chalk isn't installed

### ISSUE THREAD

**Issue by bebraw** [webpack/webpack-dev-middleware#291]
* Operating System: n/a
* Node Version: n/a
* NPM Version: n/a
* webpack version: n/a
* webpack-dev-middleware Version: n/a

- [ ] This is a **feature** request
- [x] This is a **bug**

### Code

```js
// please provide your webpack.config.js for bug reports
```

```js
// additional code
```

### Expected Behavior

Crashy McCrashster

### For Bugs; How can we reproduce the behavior?

If `writeToDisk` is enabled, the middleware tries to use `chalk`. The middleware doesn't specify it in its dependencies, though, and if the project where you are using it doesn't have it installed, it will crash the middleware.

At the very least the middleware should specify a suitable version of chalk as a peer dependency so people know to install it.

**Comment by shellscape** [comment]
@bebraw the module [depends on](https://github.com/webpack/webpack-dev-middleware/blob/master/package.json#L42) `webpack-log` which [depends on](https://github.com/webpack-contrib/webpack-log/blob/master/package.json#L32) `chalk`, so it should be there for free. You're right in that there should be a `peerDep` but I'm not sure this can throw an error in the current setup. Could it be another cause?

**Comment by bebraw** [comment]
@shellscape I can see `webpack-log`. Since I'm using npm5, it pulls `chalk` inside `webpack-log` directory, not `node_modules` root. Maybe that's why `webpack-dev-middleware` doesn't find it.

I can likely set up a tiny demo tomorrow.

**Comment by shellscape** [comment]
Interesting. Something else must be requiring a newer/older version. No need for the demo. I'll remedy this in the package.json and ping you for review.

**Comment by davidspiess** [comment]
I don't like this change.
Now all users have to manually install a color library as peer dependency, because of one cyan console statement?

```js
log.debug(chalk`{cyan Asset written to disk}: ${relativePath}`);
```

Can't you just remove chalk altogether or pull it in as normal dependency?

**Comment by shellscape** [comment]
@davidspiess chalk is already there due to the dependency on `webpack-log`. Sorry, but that won't be changed. Please feel free to fork and remove that for your needs.

**Comment by davidspiess** [comment]
@shellscape sorry but i can't understand your reasoning?

Adding it as peer dependency requires all your users to install and update a library, which they don't even know about. If you add it as normal dependency as you did in `webpack-log`, you solve @bebraw's error, and we user don't need to worry about mantaining chalk in our package.json.

**Comment by thomaswelton** [comment]
This package imports chalk here https://github.com/webpack/webpack-dev-middleware/blob/8a0bb32ba1b066dff20c51735bbaabc809a7ad72/lib/fs.js#L5

This isn't really any different from your other dependencies and shouldn't be a peer dep.
Also the inline es lint rule should not have been added, the linter was warning you about this bug.

**Comment by shellscape** [comment]
Resolved in v3.1.2. Please open a new issue if you'd like to continue discussing this, as the original issue as reported by @bebraw has been resolved.
